### ISSUE TITLE
Tensor printing should display requires_grad

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#6768]
I think one or more of the following would be nice:
- tensor printing displays requires_grad
- tensor printing displays "if this tensor is a leaf node" (but find a better name for this). Most of the times when I want to know if my tensor requires_grad I want to know if it gradients will accumulate in this.

cc @li-roy @gchanan @SsnL
